control-plane/api-gateway/controllers/gateway_controller.go

```diff
@@ -777,11 +802,80 @@ func (c *GatewayController) getRelatedHTTPRoutes(ctx context.Context, gateway ty
 
 	for _, route := range list.Items {
 		resources.ReferenceCountHTTPRoute(route)
+
+		_, err := c.getExternalFiltersForHTTPRoute(ctx, route, resources)
+		if err != nil {
+			c.Log.Error(err, "unable to list HTTPRoute ExternalFilters")
+			return nil, err
+		}
 	}
 
 	return list.Items, nil
 }
 
+func (c *GatewayController) getExternalFiltersForHTTPRoute(ctx context.Context, route gwv1beta1.HTTPRoute, resources *common.ResourceMap) ([]interface{}, error) {
+	var externalFilters []interface{}
+	for _, rule := range route.Spec.Rules {
+		ruleFilters, err := c.filterFiltersForExternalRefs(ctx, route, rule.Filters, resources)
+		if err != nil {
+			return nil, err
+		}
+		externalFilters = append(externalFilters, ruleFilters...)
+
+		for _, backendRef := range rule.BackendRefs {
+			backendRefFilter, err := c.filterFiltersForExternalRefs(ctx, route, backendRef.Filters, resources)
+			if err != nil {
+				return nil, err
+			}
+
+			externalFilters = append(externalFilters, backendRefFilter...)
+		}
+	}
+
+	return externalFilters, nil
+}
+
+func (c *GatewayController) filterFiltersForExternalRefs(ctx context.Context, route gwv1beta1.HTTPRoute, filters []gwv1beta1.HTTPRouteFilter, resources *common.ResourceMap) ([]interface{}, error) {
+	var externalFilters []interface{}
+
+	for _, filter := range filters {
+		var externalFilter client.Object
+
+		//check to see if we need to grab this filter
+		if filter.ExtensionRef == nil {
+			continue
+		}
+		switch kind := filter.ExtensionRef.Kind; kind {
+		case v1alpha1.RouteRetryFilterKind:
+			externalFilter = &v1alpha1.RouteRetryFilter{}
+		case v1alpha1.RouteTimeoutFilterKind:
+			externalFilter = &v1alpha1.RouteTimeoutFilter{}
+		default:
+			continue
+		}
+
+		//get object from API
+		err := c.Client.Get(ctx, client.ObjectKey{
+			Name:      string(filter.ExtensionRef.Name),
+			Namespace: route.Namespace,
+		}, externalFilter)
+
+		if err != nil {
+			if k8serrors.IsNotFound(err) {
+				c.Log.Info(fmt.Sprintf("externalref %s:%s not found: %v", filter.ExtensionRef.Kind, filter.ExtensionRef.Name, err))
+				//ignore, the validation call should mark this route as error
+			} else {
+				return nil, err
+			}
+		}
+
+		//add external ref (or error) to resource map for this route
+		resources.AddExternalFilter(*filter.ExtensionRef, externalFilter.(v1alpha1.ExternalRouteFilter))
+		externalFilters = append(externalFilters, externalFilter)
```

Comment:
I think you'll want a `continue` here to just not insert a blank filter into the resource map. That way the filter won't be found when you try and validate its presence in the binding logic (Sorry, the diff looks a bit bad, but in case you can't find it, it's inserted under the comment starting with `//ignore`)

```suggestion
			if k8serrors.IsNotFound(err) {
				c.Log.Info(fmt.Sprintf("externalref %s:%s not found: %v", filter.ExtensionRef.Kind, filter.ExtensionRef.Name, err))
				//ignore, the validation call should mark this route as error
				continue
			} else {
				return nil, err
			}
		}

		//add external ref (or error) to resource map for this route
		resources.AddExternalFilter(*filter.ExtensionRef, externalFilter.(v1alpha1.ExternalRouteFilter))
		externalFilters = append(externalFilters, externalFilter)
```

Reply:
Good catch, definitely meant for there to be a continue there.